lib/handle/link-reference.js

```diff
@@ -30,7 +30,12 @@ export function linkReference(node, _, context) {
   context.stack = stack
   exit()
 
-  if (type === 'full' || !text || text !== reference) {
+  if (
+    type === 'full' ||
+    !text ||
+    (text !== reference &&
+      node.children.every((node) => node.type !== 'inlineCode'))
```

Comment:
```suggestion
      node.children.every((node) => node.type === 'text'))
```

This makes it less naive and it might work?